hypervisor/boot/sbl/multiboot.c

```diff
@@ -54,7 +54,7 @@ static void parse_other_modules(struct acrn_vm *vm, const struct multiboot_modul
 			static int32_t copy_once = 1;
 
 			start = end + 1; /*it is fw name for boot args */
-			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%x ", start, mod_size, mod_addr);
+			snprintf(dyn_bootargs, 100U, " %s=0x%x@0x%lx ", start, mod_size, mod_addr);
```

Comment:
@kbillore  After check ACRN code of the function snprintf and C library standard, "%llx" is needed for "mod_addr", BTW, there is implicit type conversion (from "void *" to "unsigned long long int"), Can you make this type conversion explicitly? Can you update your commit message to include more detail? Thanks!

Reply:
@wuxyintel,  I type caste mod_add to uint64_t explicitly in https://github.com/projectacrn/acrn-hypervisor/pull/2994 and filled suppression list but as per other reviewers justification for cast was not enough, so modified %x to %lx and avoided type cast to comply with MISRA C STANDARD.

Reply:
@kbillore Can you update this PR according to previous comments? BTW, the type cast is always here either for "%x" or "%lx" or "%llx". Sorry, I don't understand how to avoid type cast. Can you explain more about this?